var/spack/repos/builtin/packages/py-petsc4py/ldshared.patch

```diff
@@ -7,7 +7,7 @@ index 8a2466a5bd..73c08b923a 100644
          ldflags = getenv('LDFLAGS', cflags + ' ' + (ldflags or ''))
          ldcmd = split_quoted(ld) + split_quoted(ldflags)
 -        ldshared = [flg for flg in split_quoted(ldshared) if flg not in ldcmd]
-+        ldshared = [flg for flg in split_quoted(ldshared) if flg not in ldcmd and (flg.find('/lib/spack/env')<0)]
++        ldshared = [flg for flg in split_quoted(ldshared) if flg not in ldcmd and (flg.find('/libexec/spack')<0)]
```

Comment:
make this worth with 1.0 and earlier versions -- then upstream it.

Reply:
See 8b7a7462f0132ed2a59803c1fab41ee5aff95975 I didn't upstream it yet (though I'd say it would be better if they find a permanent solution that doesn't require hard-coding paths)